blockprod/src/lib.rs

```diff
@@ -89,3 +89,78 @@ pub async fn make_blockproduction(
 
     Ok(Box::new(result))
 }
+
+#[cfg(test)]
+mod tests {
+    use chainstate::{ChainstateConfig, ChainstateHandle, DefaultTransactionVerificationStrategy};
+    use chainstate_storage::inmemory::Store;
+    use common::chain::{config::create_unit_test_config, ChainConfig};
+    use mempool::{MempoolHandle, MempoolSubsystemInterface};
+    use subsystem::Manager;
+
+    use super::*;
+
+    #[allow(clippy::unused_async)]
+    pub async fn setup_blockprod_test(
+    ) -> (Manager, Arc<ChainConfig>, ChainstateHandle, MempoolHandle) {
```

Comment:
What's the business with `unused_async` here? Would it be possible to just remove `async`?

Reply:
It's complaining because we don't actually do any async work in here, but from where it's called, it definitely needs it. I tried removing it and funnily enough Cargo tells me to add the `async` keyword in

Reply:
This is probably because `.await` is used. If you remove async from here and `.await` from all the callers, this won't be needed.

Reply:
Can you remove `.await` from the call sites? I don't see why it's needed there

Reply:
Gah! I know I sprinkled them all here for some reason, but yes it works without those await tails. Removed. Nice!